db/migrate/20230719080900_fix_ubuntu_versions.rb

```diff
@@ -0,0 +1,19 @@
+class FixUbuntuVersions < ActiveRecord::Migration[6.0]
+  def up
+    # For every Ubuntu operating system with a major version and a minor version in the format "<digits>(.<digits>)",
+    #   * We check if there is already an operating system present with a major version of
+    #     "<major>.<first_place_of_minor>" and a minor version of "<second_place_of_minor_or_empty>".
+    #   * If not, we update the major version of the operating system to "<major>.<first_place_of_minor>" and the minor
+    #     version to "<second_place_of_minor_or_empty>".
+    Operatingsystem.where(name: 'Ubuntu')
+      .where('operatingsystems.major ~* ?', '^\d*$')
+      .where('operatingsystems.minor ~* ?', '^\d*.?\d*$')
+      .where('o3.id IS NULL')
+      .joins("left outer join operatingsystems o3 " \
+              "on o3.name like operatingsystems.name " \
+              "and o3.major like concat(operatingsystems.major, '.', split_part(operatingsystems.minor, '.', 1)) " \
+              "and o3.minor like split_part(operatingsystems.minor, '.', 2)")
```

Comment:
I thinkk you can use `=` here instead of `like`, since you do not compare with a wildcard-string.

Reply:
Thanks for having a look at the migration and your input, @m-bucher :)